app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -13,13 +13,15 @@ class << model_name
     end
 
     has_many :availability_zones,            :foreign_key => :ems_id, :dependent => :destroy
-    has_many :flavors,                       :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_tenants,                 :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_resource_quotas,         :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_volumes,                 :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_volume_snapshots,        :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_object_store_containers, :foreign_key => :ems_id, :dependent => :destroy
     has_many :cloud_object_store_objects,    :foreign_key => :ems_id, :dependent => :destroy
+    has_many :cloud_databases,               :foreign_key => :ems_id, :dependent => :destroy
+    has_many :database_flavors,              :foreign_key => :ems_id, :dependent => :destroy, :class_name => 'DatabaseFlavor'
+    has_many :flavors, -> { where.not(:type => DatabaseFlavor.descendants) }, :foreign_key => :ems_id, :dependent => :destroy
```

Comment:
@Fryguy We thought about introducing an intermediate `VmFlavor` or `InstanceFlavor`, then all providers `Flavor` classes need to be derived from it.

Solution 1:
Rename all provider specific Flavors, for example, we will have
```
ManageIQ::Providers::Amazon::CloudManager::VmFlavor < ::VmFlavor < ::Flavor
ManageIQ::Providers::Amazon::CloudManager::CloudDbFlavor < ::CloudDbFlavor < ::Flavor
```
and base `CloudManager` will
```
have_many :vm_flavors
have_many :cloud_db_flavors
```
I am not sure the impact on existing code, specially the reporting.

Solution 2
Still use the name flavor for VMs, and use cloud_db_flavor for db instances, for example
```
ManageIQ::Providers::Amazon::CloudManager::Flavor < ::VmFlavor < ::Flavor
ManageIQ::Providers::Amazon::CloudManager::CloudDbFlavor < ::CloudDbFlavor < ::Flavor
```
and base `CloudManager` will
```
have_many :flavors :class_name => 'VmFlavor'
have_many :cloud_db_flavors :class_name => `CloudDbFlavor`
```
With this solution all flavor related existing code still works, but it is strange to see the hierarchy of vm flavors.

Solution 3:
All provider specific flavors are still called flavor but under its own name space.
```
ManageIQ::Providers::Amazon::CloudManager::Vm::Flavor < ::VmFlavor < ::Flavor
ManageIQ::Providers::Amazon::CloudManager::CloudDatabase::Flavor < ::CloudDbFlavor < ::Flavor
```
and base `CloudManager` will
```
have_many :vm_flavors
have_many :cloud_db_flavors
```
With this solution we still need to rename existing provider flavor class, and change the has_many relations for base `CloudManager`

Any preference?